io/src/pcd_io.cpp

```diff
@@ -1634,35 +1624,68 @@ pcl::PCDWriter::writeBinaryCompressed (const std::string &file_name, const pcl::
     }
   }
 
-  char* temp_buf = static_cast<char*> (malloc (static_cast<size_t> (static_cast<float> (data_size) * 1.5f + 8.0f)));
+  std::vector<char> temp_buf (data_size * 3 / 2 + 2 * sizeof (unsigned int));
   // Compress the valid data
-  unsigned int compressed_size = pcl::lzfCompress (only_valid_data, 
+  unsigned int compressed_size = pcl::lzfCompress (&only_valid_data.front (), 
```

Comment:
Informative note, std::vector exposes a .data() method which does exactly this.

Reply:
Good catch.

...another C++11 improvement I seem to have missed.  I've long used std::string::data(), when I don't need null-termination.  Nice to see it in std::vector.

Feel free to change it, if you want.

Reply:
Oh right, it's a C++11 feature, so we can not integrate it anyway.